pkg/cmd/cli/cmd/importimage.go

```diff
@@ -157,20 +168,56 @@ func (o *ImportImageOptions) Run() error {
 	case err != nil:
 		return err
 	default:
-		if wasError(result) {
-			fmt.Fprintf(o.errout, "The import completed with errors.\n\n")
+		if o.DryRun {
+			if wasError(result) {
+				fmt.Fprintf(o.errout, "The dry-run import completed with errors.\n\n")
+			} else {
+				fmt.Fprint(o.out, "The dry-run import completed successfully.\n\n")
+			}
 		} else {
-			fmt.Fprint(o.out, "The import completed successfully.\n\n")
+			if wasError(result) {
+				fmt.Fprintf(o.errout, "The import completed with errors.\n\n")
+			} else {
+				fmt.Fprint(o.out, "The import completed successfully.\n\n")
+			}
 		}
 
-		// optimization, use the image stream returned by the call
-		d := describe.ImageStreamDescriber{Interface: o.osClient}
-		info, err := d.Describe(o.Namespace, stream.Name, kctl.DescriberSettings{})
-		if err != nil {
-			return err
+		if result.Status.Import != nil {
+			// TODO: dry-run doesn't return an image stream, so we have to display partial results
+			info, err := describe.DescribeImageStream(result.Status.Import)
+			if err != nil {
+				return err
+			}
+			fmt.Fprintln(o.out, info)
 		}
 
-		fmt.Fprintln(o.out, info)
+		if repo := result.Status.Repository; repo != nil {
+			for _, image := range repo.Images {
+				if image.Image != nil {
+					info, err := describe.DescribeImage(image.Image, imageapi.JoinImageStreamTag(stream.Name, image.Tag))
```

Comment:
I'm not sure we want to print that much of an information after import. Imagine how this will look like when we're importing ~90 tags, see https://bugzilla.redhat.com/show_bug.cgi?id=1433244 for more info. It will be completely unreadable. The same applies to below resutl.Status.Images. You want that information use oc describe, oc get.